# **Kelompok 6 Mentor Jonatan Situmorang**
# Anggota:
# - Adam Muslim
# - Naufal Maulana Faza
# - Viona Siahaan

# Melalui dataset dari Kaggle disini kami akan melakukan analisis data Jumlah Produksi dengan model Linear Regresion.

# Bagian 1: Viona

# * Import Packages dan Dataset

Pada bagian ini kita akan mengimpor library untuk menganalisis dan memvisualisasi data, mengatur plot, dan mengaktifkan mode plot inline.
- NumPy adalah library yang digunakan untuk komputasi numerik di Python. Ini menyediakan objek array multidimensi yang efisien, dengan berbagai fungsi untuk bekerja dengan array.
- Pandas adalah library yang digunakan untuk analisis data di Python. Ini menyediakan struktur data dan fungsi untuk memanipulasi data tabular dan deret waktu.
- Matplotlib adalah library yang digunakan untuk membuat visualisasi data dalam bentuk grafik, plot, dan visualisasi lainnya di Python.
-  Seaborn adalah library Python yang dibangun di atas Matplotlib dan digunakan untuk membuat plot statistik yang menarik dan informatif.

In [ ]:
#library import
import numpy as np
import pandas as pd
from pandas import Series,DataFrame #dari Pandas mengimport "Series" struktur data array atau "DataFrame" struktur data dari tabel database

#packages for plotting
import matplotlib.pyplot as plt
import seaborn as sns
sns.set_style('whitegrid') #menampilkan latar belakang putih dengan garis-garis grid.

#plot akan ditampilkan di output notebook secara otomatis setelah dihasilkan tanpa memerlukan pemanggilan tambahan ke fungsi 'plt.show()'.
%matplotlib inline

Selanjutnya kita akan menginstal paket openpyxl yang akan digunakan untuk memanipulasi file excel. Dengan paket ini kita dapat membaca, mengedit, dan membuat file excel dengan Python.

In [ ]:
pip install openpyxl

     |████████████████████████████████| 242 kB 922 kB/s            
Note: you may need to restart the kernel to use updated packages.


Selanjutnya kita akan melakukan pembacaan file excel yang telah kita ambil, untuk ditampilkan baris pertama dari data sebagai pemeriksaan awal.

In [ ]:
#dari Pandas untuk membaca file Excel yang diberikan dengan 'index_col=0' untuk mengatur kolom pertama sebagai indeks baris dari DataFrame.
data = pd.read_excel("../input/data-regresi-linier/data.xlsx", index_col=0)

#menampilkan beberapa baris pertama dari DataFrame yang telah dibaca.
data.head()

Jumlah_Produksi  Luas_Lahan  Jumlah_Pohon  Pupuk_Kandang  \
Responden                                                             
1                      200         900            20             50   
2                      500         500            50            250   
3                      500         500            50            250   
4                      900         900            60            450   
5                      900         900            60            450   

           Pupuk_Phonska  Pestisida  Tenaga_Kerja  
Responden                                          
1                     10        600            27  
2                     25       1600            21  
3                     25       1600            33  
4                     45       3600            40  
5                     45       3600            50

# Bagian 2: Naufal

# * Import scikit learn untuk mengestimasi model regresi

# Pada bagian ini berisi tentang import scikit dan model regresi linear

Scikit-Learn atau yang dikenal juga dengan sklearn merupakan library Python yang bersifat open-source. Library ini akan membantu Python untuk mengakses berbagai algoritma machine learning dan alat untuk melakukan tugas-tugas umum dalam Data Science.

Regresi linear adalah teknik analisis data yang memprediksi nilai data yang tidak diketahui dengan menggunakan
nilai data lain yang terkait dan diketahui. Secara matematis memodelkan variabel yang tidak diketahui atau tergantung dan variabel yang dikenal atau independen sebagai persamaan linier.

In [ ]:
# Import for Linear Regression
import sklearn
from sklearn.linear_model import LinearRegression

lreg = LinearRegression()

# 1. set variabel x dan y

In [ ]:
# Data Columns
X_multi = data.drop('Jumlah_Produksi',1)

# Targets
Y_target = data.Jumlah_Produksi

/opt/conda/lib/python3.7/site-packages/ipykernel_launcher.py:2: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only
  


In [ ]:
lreg.fit(X_multi,Y_target)

LinearRegression()

# Bagian 3: Adam

# 2. hasil estimasi koefisien intercept dan x

Menampilkan perkiraan koefisien intersep

In [ ]:
print(' The estimated intercept coefficient is %.2f ' %lreg.intercept_)

 The estimated intercept coefficient is 8.57 


Menampilkan banyaknya koefisien yang digunakan

In [ ]:
print(' The number of coefficients used was %d ' % len(lreg.coef_))

 The number of coefficients used was 6 


In [ ]:
# Set a DataFrame from the Features
# Tetapkan DataFrame dari Fitur
coeff_df = DataFrame(data.columns)
coeff_df.columns = ['Features']

# Set a new column lining up the coefficients from the linear regression
# Tetapkan kolom baru yang menyusun koefisien dari regresi linier
coeff_df["Coefficient Estimate"] = pd.Series(lreg.coef_)

# Show
# Menunjukkan
coeff_df

Features  Coefficient Estimate
0  Jumlah_Produksi             -0.002541
1       Luas_Lahan              1.575790
2     Jumlah_Pohon              0.610595
3    Pupuk_Kandang              4.542999
4    Pupuk_Phonska              0.080771
5        Pestisida              0.328858
6     Tenaga_Kerja                   NaN

Model regresi yang didapat :
Jumlah_Produksi = 8.57 + 1.575790Luas_Lahan + 0.610595Jumlah_Pohon + 4.542999Pupuk_Kandang + 0.080771Pupuk_Phonska + 0.328858Pestisida

# **Bentuk umum Regresi Linier Berganda**
## Y = a + b1 X1 + b2 X2 + .... + bn Xn
# **Keterangan:**

*   Y adalah variabel terikat/dependent
*   a adalah konstanta (intersept)
*   b adalah koefisien regresi pada masing-masing variabel bebas.
*   X adalah variabel-variabel bebas/independent
*   n adalah banyaknya variabel x



# **author: @febymelania**
# **Link Sumber Public Project Kaggle:**
# Dataset: https://www.kaggle.com/datasets/febymelania/data-regresi-linier/code
# Code: https://www.kaggle.com/code/febymelania/regresi-linier-jumlah-produksi